### PR TITLE
#4155 [Manager] Pipeline cannot be completely removed with "Delete this pipeline" menu from Toolbar

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4155)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js [hunk 1]
```diff
@@ -29,6 +29,20 @@ const pipeline = () => {
...
   const updateCells = paperApi => {
+    // TODO: this is a temporary fix for "users cannot completely delete a pipeline" via
+    // our UI. This is due to that the action also triggers Paper's onChange event where
+    // after the action removes the last element from the Paper, the onChange event would
+    // fire and so a PUT request is made to the backend, therefore, the deleted pipeline is back again.
+    // A more comprehensive solution:
+    // In 0.10, we will rewrite the "delete this pipeline" action without using `pipelineApiHelper`
+    // which can completely avoid firing Paper's onChange event. The issue can be found here:
+    // https://github.com/oharastream/ohara/issues/4149
+    const isDeletingPipeline = !window.location.pathname.includes(
```
Comment (saivirtue) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: hmm...interesting condition.
What happened for the following steps?
1. create pipelines: myppp and pp
2. switch to `/workspace1/pp` and **delete this pipeline**
3. UI redirect to `/workspace1/myppp` and ?

Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: Good point @saivirtue Let's test this case :)

File: ohara-manager/client/src/components/Pipeline/Toolbar/ToolbarHooks.js [hunk 2]
```diff
@@ -17,47 +17,52 @@
...
-  const { remove: removeTopic } = pipelineUtils.topic();
+  const { deleteConnector, stopConnector } = useConnectorActions();
+  const { deleteTopic } = useTopicActions();
+  const { deleteStream, stopStream } = useStreamActions();
 
   const deleteCells = async cells => {
-    setSteps([...cells.map(cell => cell.name)]);
+    const cellNames = cells.map(cell => {
+      if (cell.kind !== KIND.topic) return cell.name;
+      return cell.isShared ? cell.name : cell.displayName;
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Toolbar/ToolbarHooks.js]: pipeline-only topics should use the display name

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js (hunk 1 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: @saivirtue It does have issue like you mention in the above. I've written the delete logic without using paperApiHelper which effectively avoid the onChange event issue as well as getting rid of the hacky code I previously purposed :)

**Comment by konekoya** [comment]
retry manager-e2e

**Pull request merged**